lib/eco/treelexer/lexer.py

```diff
@@ -257,3 +309,27 @@ def lex(self, text):
                     result.append((text[self.pos:], None, 0))
                 break
         return result
+
+    def treelex(self, node):
+        pm = TreePatternMatcher()
+        pos = 0
+        result = []
+        while True:
+            lookahead = 0
+            oldpos = pos
+            oldnode = node
+            for p, n in self.patterns:
+                pm.reset()
+                token = pm.match(p, node)
+                if token:
+                    lookahead = lookahead + (1 if not pm.exactmatch else 0)
```

Comment:
Let's please not use Python's awful ternary operator (which isn't left-to-right evaluated) -- an If statement would only be 1 line longer ;)